src/schema_helper.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {z} from 'zod';
+import {zodToJsonSchema} from 'zod-to-json-schema';
+
+import {Schema, Type} from './types';
+
+export function zodToGoogleGenAISchema(
+  vertexai: boolean,
+  schema: z.ZodObject<z.ZodRawShape>,
+): Schema {
+  const jsonSchema = zodToJsonSchema(schema, 'zodSchema').definitions![
```

Comment:
Why not use the openApi3 target to help with conversion?
```ts
zodToJsonSchema(zodType, { target: 'openApi3' })
```

Afaik, it’s the required target, and it probably can reduce the processing needed.